### PR TITLE
Add left/right shift operator support

### PR DESCRIPTION
**Pull request by yetingk**
I am sorry about too many commits for minor modification.

### CODE REVIEW

File: parser.c [hunk 1]
```diff
@@ -889,7 +901,8 @@ static Ast *read_if_stmt(void)
     expect(')');
     Ast *then = read_stmt();
     Token *tok = read_token();
-    if (!tok || get_ttype(tok) != TTYPE_IDENT || strcmp(get_ident(tok), "else")) {
```
Comment (jserv) [inline parser.c:892]: Don't touch this line since there is no functional change.

File: tests/arith.c [hunk 2]
```diff
@@ -69,6 +75,6 @@ int main()
     test_ternary();
     test_logand();
     test_bitand();
-
```
Comment (jserv) [inline tests/arith.c:72]: No blank line between `test_bitand` and `test_shift`.

File: codegen_x64.c [hunk 3]
```diff
@@ -327,6 +333,9 @@ static void emit_binop_int_arith(Ast *ast)
     if (ast->type == '/') {
         emit("mov $0, %%edx");
         emit("idiv %%rcx");
+    } else if (ast->type == PUNCT_LSHIFT ||
```
Comment (jserv) [inline codegen_x64.c]: Use `clang-format -i` to ensure consist style.

**Comment by jserv** [comment]
This pull request looks good at first glance. However, signedness is not yet supported in latest MazuCC, and it impractical to distinguish between logical and arithmetic shifts. I would expect there is remark in code generation.

File: parser.c (hunk 1 above)
Comment (yetingk) [reply parser.c:892]: But this change is made by command "clang-format -i *.[ch]". Should I maintain the style in ordinary file? (There also exist changes in lexer.c and mzcc.h).

Comment (jserv) [reply parser.c:892]: I suffered from style inconsistency problem by different versions of `clang-format`. The version I am using is `8.0.0`, and I would suggest to minimize the changes.

**Comment by jserv** [comment]
@fakepaper56, any further changes?

**Pull request merged**